File: FrontEnd/src/app/features/reservations/users/classes/dtos/user-read-dto.ts
import { Guid } from 'guid-typescript'
// Custom
import { CustomerAutoCompleteVM } from '../../../customers/classes/view-models/customer-autocomplete-vm'
import { Metadata } from 'src/app/shared/classes/metadata'

export interface UserReadDto extends Metadata {

    id: Guid
    username: string
    displayname: string
    customer: CustomerAutoCompleteVM
    email: string
    isFirstFieldFocused: boolean
    isAdmin: boolean
    isActive: boolean

}

File: API/Infrastructure/Users/Dtos/UserReadDto.cs
using API.Infrastructure.Classes;
using API.Infrastructure.Interfaces;

namespace API.Infrastructure.Users {

    public class UserReadDto : IMetadata {

        // Fields
        public string Id { get; set; }
        public string Username { get; set; }
        public string Displayname { get; set; }
        public bool IsFirstFieldFocused { get; set; }
        public SimpleEntity Customer { get; set; }
        public string Email { get; set; }
        public bool IsAdmin { get; set; }
        public bool IsActive { get; set; }
        // Metadata
        public string PostAt { get; set; }
        public string PostUser { get; set; }
        public string PutAt { get; set; }
        public string PutUser { get; set; }

    }

}